corehq/apps/saved_reports/scheduled.py

```diff
@@ -95,14 +95,25 @@ def get_scheduled_report_ids(period, start_datetime=None, end_datetime=None):
             # Don't care if not on hour
             continue
         keys = _make_all_notification_view_keys(period, target_point_in_time)
+
         for key in keys:
             for result in ReportNotification.view(
                 "reportconfig/all_notifications",
                 reduce=False,
                 include_docs=False,
                 **key
             ).all():
-                yield result['id']
+                if period == 'hourly':
+                    try:
+                        hour = int(result['value'].get('hour'))
+                        stop_hour = int(result['value'].get('stop_hour'))
```

Comment:
If `hour` or `stop_hour` are None (resulting in a TypeError) then this `try...except` block will treat their values as 0 and 23. That seems right. They can't be strings because they are stored as `IntegerProperty`. So I think if you use explicit defaults, you could drop the `try...except`.
```suggestion
                        hour = int(result['value'].get('hour') or 0)
                        stop_hour = int(result['value'].get('stop_hour') or 23)
```

Reply:
Seems reasonable